### DISCUSSION TITLE
Are tags with spaces supported?

### DISCUSSION THREAD

**Discussion by wickertongue** [vkbo/novelWriter#1194] [Help] [answered]
Hi folks, am loving novelWriter but have a question / issue:

1. Say I have a file called 'History of Saturn' under the 'Timeline' directory.
2. In the 'History of Saturn' file I add a tag: `@tag: History of Saturn`
![Screenshot from 2022-10-19 21-14-16](https://user-images.githubusercontent.com/52103420/196795206-b4b223a7-1a11-4cb3-a894-50213ba9be61.png)

3. I then reference that tag elsewhere in my notes: `@time: History of Saturn`
![Screenshot from 2022-10-19 21-15-15](https://user-images.githubusercontent.com/52103420/196795204-ab754c47-5d54-48ef-93af-d31b819f5127.png)

There are no wiggly lines, and the tag seems to be reference correctly, but I can't follow that tag. Instead, I get: `Could not find the reference for tag 'Saturn'. It either doesn't exist, or the index is out of date. The index can be updated from the Tools menu, or by pressing F9.`

![Screenshot from 2022-10-19 21-15-59](https://user-images.githubusercontent.com/52103420/196795777-df38ba9a-77a3-40c2-9418-62a4f26fe2af.png)

Does novelWriter support tags with spaces? Or should I use camelcase tags with no spaces?

**Comment by vkbo** [comment, marked as answer]
The algorithm that processes tags only splits by comma, and doesn't care what is between commas, including spaces (although it ignores leading and trailing white spaces).

However, there is one exception, which is when you use the `Ctrl+Enter` shortcut to follow a tag in the editor, only the word under the cursor is picked up. You can clearly see this from the error message. Here the code uses some internal logic from the Qt library. That is certainly inconsistent, and should probably be fixed. I will create an issue ticket from this.

I rewrote the part of the code that lets you follow tags a while back, and may have forgotten about spaces in the process.

TLDR: Yes, spaces should be allowed. This is a bug.
